eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/eventlist/view/FlexibleAdapterEventListView.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.eventlist.view;
+
+import android.support.v7.widget.RecyclerView;
+
+import com.schedjoules.eventdiscovery.eventlist.itemsprovider.EventListItemsProviderImpl;
+
+import eu.davidea.flexibleadapter.FlexibleAdapter;
+import eu.davidea.flexibleadapter.items.IFlexible;
+
+
+/**
+ * Handles the RecyclerView and the Adapter initialization for the Event list.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class FlexibleAdapterEventListView implements EventListView
```

Comment:
I agree that it's a rather unfortunate class like this and even the responsibility is not clear.
The purpose of both `EventListView` and `EventListToolbar` is to extract the low level UI initialization (and possibly controlling) code from the Fragment. They serve as kind of the View or ViewController in an MVP like thinking, so they serve as an handler for the UI, similar as the ScreenView was in the first version of the Activity. With the Toolbar, it's more clear I think, we might even say that we declare a Toolbar, with the EventListView, because of the adapter, and also because many related loading logic is not in this class, it is not really good, I agree.

The question is what would be a better way? Leaving everything in the Fragment/Activity that we couldn't find a very good object representation of leads to build up of various low-level code there, growing the classes too big and making it harder to read those controller level logics.
So the question from the other side is what is the Fragment's responsibility? We know it's sort of corrupted by default, but it shouldn't mean we give up on it and pour everything in it. I think what they have to do as a minimum in our current usage is the initialization of the components and acting as presentation layer controller to handle the high level callbacks, both the lifecycles and the ones from our own components. I think it would be best if everything else is delegated, and it doesn't contain low-level code.
So if we move those code under discussion now back to the Fragment, then it will contain methods like this:
```
    @Override
    public void onLocationSelected(LocationSelectionResult result)
    {
        mLastSelectedLocation.update(result);
        mToolbar.setToolbarTitle(result.name());
        mListItemsProvider.loadEvents(result.geoLocation(), startAfter());
    }
```
```
private void setupToolabr()
    {
        mToolbar = toolbar;

        mToolbar.setTitle(""); // Need to set it to something here, otherwise it will use title from Activity label
        mToolbar.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                listener.onToolbarTitleClick();
            }
        });

        BaseActivity activity = (BaseActivity) fragmentActivity;
        activity.setSupportActionBar(mToolbar);

        Resources res = activity.getResources();
        if (res.getBoolean(R.bool.schedjoules_enableBackArrowOnEventListScreen))
        {
            //noinspection ConstantConditions
            activity.getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }
        else
        {
            mToolbar.setTitleMarginStart(res.getDimensionPixelSize(R.dimen.schedjoules_list_item_horizontal_margin));
        }
    }
```
They have clearly different level of abstractions, the first one is high-level, delegating, the second one low-level. Why would a high-level orchestrator component need to know and deal with what is the margin of the toolbar and what extra framework boilerplate is needed to wire things together?

So my main concern is that I think we should keep the Fragment clean as well, not just the objects we create and I don't see a better option currently, although there might be of course.
So I propose moving `EventListView` code back to Fragment but keeping `EventListToolbar` as an object that represents the Toolbar and hides its initiation logic.

Let me know how you would refactor this, I will do that.